lib/reindex/s3_bucketd.py

```diff
@@ -204,9 +229,11 @@ def get_next_marker(p):
 
             if buckets:
                 yield buckets
-                if name is not None:
-                    # Break on the first matching bucket if a name is given
-                    break
+                if names:
+                    seen_buckets.update(b.name for b in buckets)
+                    # Break if we have seen all the buckets we are looking for
+                    if all(b in seen_buckets for b in names):
+                        break
```

Comment:
Minor suggestion: if we know that `names` doesn't have duplicate bucket names (or alternatively we could ensure it doesn't), it should be enough to check that the size of the set is equal to `len(names)`.

Reply:
While fiddling with this I realized that with the addition of `get_bucket_md` the `names` param isn't used anywhere. I've simplified the function to just remove it.